staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -374,10 +401,22 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return nil
 		}
 
-		return o.PrintObj(outputObj, o.Out)
+		return o.PrintObj(dataChangeMsg).PrintObj(outputObj, o.Out)
 	})
 }
 
+// updateDataChangeMsg gets final operation msg
+func updateDataChangeMsg(oldObj, newObj []byte, overwrite bool) string {
+	msg := MsgNotAnnotated
+	if !cmp.Equal(oldObj, newObj) {
+		msg = MsgAnnotated
+		if !overwrite && len(newObj) < len(oldObj) {
```

Comment:
I realize that this is what `label` is doing, but I don't think it is an effective strategy for determining what happened, and as such would rather not propagate it to additional commands. It is possible to both add and remove labels/annotations with a single command. Right now, which of the messages gets printed depends on the relative length of the labels/annotations affected. Here's an example:

```bash
$ kubectl label pod mypod a=b aa=bb
pod/mypod labeled

$ kubectl label pod mypod a- ee=ff
pod/mypod labeled # because the new label is textually longer

$ kubectl label pod mypod aa- e=f # same operations as above
pod/mypod unlabeled # different message, because the old label is textually longer
```

What message makes sense in this hybrid scenario? We could say "labelled and unlabelled", but perhaps a generic "labels modified" would be simpler. Wdyt?

Reply:
Got it, agree with the "labels modified" msg.
Updated, PTAL thanks.